xrdsst/controllers/key.py

```diff
@@ -0,0 +1,203 @@
+        security_servers = list(filter(lambda ss: ss["name"] == ss_name, config["security_server"]))
+        if len(security_servers) > 0:
+            ss_api_config = self.create_api_config(security_servers[0], config)
+            return self.remote_list_keys(ss_api_config, tokens)
+        else:
+            BaseController.log_info(self.security_server_not_found_message(ss_name))
+
+        BaseController.log_keyless_servers(ss_api_conf_tuple)
+
+    def delete_keys(self, config, ss_name, key_ids):
+        ss_api_conf_tuple = list(zip(config["security_server"],
+                                     map(lambda ss: self.create_api_config(ss, config), config["security_server"])))
+        security_servers = list(filter(lambda ss: ss["name"] == ss_name, config["security_server"]))
+        if len(security_servers) > 0:
+            ss_api_config = self.create_api_config(security_servers[0], config)
+            self.remote_delete_keys(ss_api_config, ss_name, key_ids)
+        else:
+            BaseController.log_info(self.security_server_not_found_message(ss_name))
+
+        BaseController.log_keyless_servers(ss_api_conf_tuple)
+
+    def update_keys(self, config, ss_name, key_id, friendly_name):
+        ss_api_conf_tuple = list(zip(config["security_server"],
+                                     map(lambda ss: self.create_api_config(ss, config), config["security_server"])))
+        security_servers = list(filter(lambda ss: ss["name"] == ss_name, config["security_server"]))
+        if len(security_servers) > 0:
+            ss_api_config = self.create_api_config(security_servers[0], config)
+            self.remote_update_key(ss_api_config, ss_name, key_id, friendly_name)
+        else:
+            BaseController.log_info(self.security_server_not_found_message(ss_name))
+
+        BaseController.log_keyless_servers(ss_api_conf_tuple)
+
+    def remote_list_keys(self, ss_api_config, tokens):
+        tokens_api = TokensApi(ApiClient(ss_api_config))
+        render_list = []
+        render_data = []
+        for token in tokens:
+            try:
+                token = tokens_api.get_token(token)
+                render_list.extend(self.parse_token_response_to_key_list(token))
+            except ApiException as err:
+                BaseController.log_api_error("TokensApi=>get_token", err)
+
+        if len(render_list) > 0:
+            if self.is_output_tabulated():
+                render_data = [KeyListMapper.headers()]
+                render_data.extend(map(KeyListMapper.as_list, render_list))
+            else:
+                render_data.extend(map(KeyListMapper.as_object, render_list))
+            self.render(render_data)
+
+        return render_list
+
+    @staticmethod
+    def remote_delete_keys(ss_api_config, ss_name, key_ids):
+        keys_api = KeysApi(ApiClient(ss_api_config))
+        for key_id in key_ids:
+            try:
+                keys_api.delete_key(key_id)
```

Comment:
Maybe you should also check here like in "update key", if the key exists by calling first get_key() and only if the key exists, then try to delete it